### PR TITLE
Added Contributions Best Practices

### LINKED ISSUE #1560: Documentation regarding Contribution should be Improved
Documentation regarding Contribution should be Improved because many Contributors fails to follow the Basic Ethics of Contributing.

### PR DESCRIPTION
**Pull request by naveentherocker**
Fixes #1560
Added Documentation regarding Contributing to Github.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -291,7 +291,26 @@ The web-app is able to automatically deploy the generated event sites to Github
...
+ **Commits**
+ * Write clear meaningful git commit messages (Do read http://chris.beams.io/posts/git-commit/)
+ * Make sure your PR's description contains GitHub's special keyword references that automatically close the related issue when the PR is merged. (More info at https://github.com/blog/1506-closing-issues-via-pull-requests )
+ * When you make very very minor changes to a PR of yours (like for example fixing a failing travis build or some small style corrections or minor changes requested by reviewers) make sure you squash your commits afterwards so that you don't have an absurd number of commits for a very small fix. (Learn how to squash at https://davidwalsh.name/squash-commits-git )
+ * When you're submitting a PR for a UI-related issue, it would be really awesome if you add a screenshot of your change or a link to a deployment where it can be tested out along with your PR. It makes it very easy for the reviewers and you'll also get reviews quicker.
+ 
+ **Code Styleguide**
+ * Do follow the .editorconfig file regarding maintaining of code style (It's mandatory).
+ * For more information regarding .editorconfig file, see [editorconfig](http://editorconfig.org/#download)
```
Comment (sumedh123) [inline README.md]: There is no .editorconfig here.

**Comment by naveentherocker** [comment]
Fixed It @sumedh123

**Comment by aayusharora** [comment]
screen shot 2017-12-23 at 5 28 18 pm

**Comment by naveentherocker** [comment]
Well I think it's fine now @aayusharora

**Comment by naveentherocker** [comment]
@sumedh123 @aayusharora Please

**Comment by naveentherocker** [comment]
@YashKumarVerma there are commits from two in this PR.

**Comment by aayusharora** [comment]
Hi @naveentherocker 
I can still find a lot of grammatical errors.
Will you please run Grammarly over it.
https://app.grammarly.com/docs/246135608

**Comment by naveentherocker** [comment]
@aayusharora Now I fixed everything.

**Comment by naveentherocker** [comment]
@aayusharora Please Review. I am waiting for 10 Days

**Comment by naveentherocker** [comment]
@geekyd Please Merge this PR

**Pull request merged**